services/api/src/resources/openshift/resolvers.ts

```diff
@@ -114,7 +114,22 @@ export const getOpenshiftByEnvironmentId: ResolverFn = async (
   args,
   { sqlClientPool, hasPermission }
 ) => {
-  await hasPermission('openshift', 'viewAll');
+  // get the project id for the environment
+  const projectrows = await query(
+    sqlClientPool,
+    `SELECT e.project
```

Comment:
There is a preference to reduce raw sql queries in resolvers. For permissions checks you could  re-use a knex/sql object `selectProjectByEnvironmentId` (first [import](https://github.com/uselagoon/lagoon/blob/main/services/api/src/resources/environment/resolvers.ts#L11), then use) or avoid the `query` all together by using a project helper like `getProjectByEnvironmentId` (first [import](https://github.com/uselagoon/lagoon/blob/main/services/api/src/resources/environment/resolvers.ts#L12), then [use](https://github.com/uselagoon/lagoon/blob/main/services/api/src/resources/environment/resolvers.ts#L244-L246))

Reply:
refactored with the helper